### ISSUE TITLE
Stages resulting in long build output fail to save results.

### ISSUE THREAD

**Issue by akiellor** [quad/bt#9]
Fails with the following trace. It looks like the commit message is too long.

```
/Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/process.rb:290:in `spawn': Argument list too long - /usr/bin/git (Errno::E2BIG)
    from /Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/process.rb:290:in `spawn'
    from /Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/process.rb:271:in `popen4'
    from /Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/process.rb:97:in `exec!'
    from /Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/process.rb:68:in `initialize'
    from /Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/git.rb:301:in `new'
    from /Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/git.rb:301:in `native'
    from /Users/akiellor/.rvm/gems/ruby-1.9.2-p180@bt/gems/grit-2.4.1/lib/grit/git.rb:325:in `method_missing'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:130:in `commit'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:27:in `block in workspace'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:56:in `call'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:56:in `block (2 levels) in working_tree'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:65:in `block in initialize'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:65:in `chdir'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:65:in `initialize'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:54:in `new'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:54:in `block in working_tree'
    from /Users/akiellor/.rvm/rubies/ruby-1.9.2-p180/lib/ruby/1.9.1/tmpdir.rb:83:in `mktmpdir'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:52:in `working_tree'
    from /Users/akiellor/Development/bt/lib/bt/git.rb:22:in `workspace'
    from /Users/akiellor/Development/bt/lib/bt/stage.rb:22:in `build'
    from /Users/akiellor/Development/bt/bin/bt-go:34:in `<main>'
```

**Comment by quad** [comment]
My guess is the commit message is supplied to git-commit via the command-line. (--message)

We need to pipe it in; either via standard-input, an external pipe, or an external file. (Descending order of preference.)

**Comment by akiellor** [comment]
Was fixed by #12.
